PetGame/Controllers/PetController.cs

```diff
@@ -94,9 +114,89 @@ public void Put(ulong id, [FromBody]Pet value)
         /// <param name="id"></param>
         // DELETE api/<controller>/5
         [HttpDelete("{id}")]
-        public void Delete(ulong id)
+        public IActionResult Delete(ulong id)
+        {
+            var user = loginService.GetUserFromContext(HttpContext.User);
+
+            if (user == null)
+                return Unauthorized();
+
+            if (petService.DeletePet(id, user.UserId))
+            {
+                // deleted ok
+                return Ok();
+            }
+            // didn't delete Ok, either not found or wrong user
+            return Unauthorized();
+        }
+
+        // Activity
+        // GET /api/Pet/petId/Activity
+        // POST /api/Pet/petId/ActivityOptions
+        // {
+        //"Limit": 10,
+        //"After": "2012-04-23T18:25:43Z",
+        //"Type": 'd'
+        //   }
+        [HttpGet("{petId}/Activity")]
+        [HttpPost("{petId}/ActivityOptions")] // this must not be under /Activity, because that doesn't follow rest convention for POST
```

Comment:
This allows for the creation of new activities. Again, expected behavior?

Reply:
What are you talking about? Nothing in this method creates new activities.

Reply:
When I used POST, it created new Activities in the DB. I will double check my results.

Reply:
When I used POST, it created new Activities in the DB. I will double check my results.

Reply:
This is POST Activity**Options**. You are confusing it with the Post method below this.

Reply:
Not sure what I did, but upon further tests, I cannot reproduce this behavior.